wire-ios-sync-engine/Source/UserSession/SyncStatus.swift

```diff
@@ -328,21 +340,91 @@ public extension SyncStatus {
         }
     }
 
-    private func log(_ message: String? = nil) {
-        let info = SyncStatusLog(
-            phase: currentSyncPhase.description,
-            isSyncing: isSyncing,
-            pushChannelEstablishedDate: pushChannelEstablishedDate?.description,
-            message: message
-        )
-        do {
-            let data = try JSONEncoder().encode(info)
-            let jsonString = String(decoding: data, as: UTF8.self)
-            let message = "SYNC_STATUS: \(jsonString)"
-            WireLogger.sync.info(message, attributes: .safePublic)
-        } catch {
-            let message = "SYNC_STATUS: \(description)"
-            WireLogger.sync.error(message, attributes: .safePublic)
+    private func didCompleteSyncPhase(
+        _ phase: SyncPhase,
+        completedAllPhases: Bool
+    ) {
+        let currentTime = Date.now
+        let phaseStartTime = syncTimeTracker.phaseStartTime
+        let duration = currentTime.timeIntervalSince(phaseStartTime)
+
+        logSyncPhaseCompleted(phase: phase, duration: duration)
+        syncTimeTracker.addPhaseDuration(duration)
+        // resetting for next sync phase
+        syncTimeTracker.resetStartTime()
+
+        if completedAllPhases {
+            logSyncCompleted()
+            // Sync is completed and logged, resetting tracked time values
+            resetSyncTimeTracker()
         }
     }
+
+    func resetSyncTimeTracker() {
+        syncTimeTracker.reset()
+    }
+}
+
+// MARK: - Logging
+
+extension SyncStatus {
+    
+    /// Logs the initial / incremental sync start
+    private func logSyncStarted() {
+        let message = "did start \(isSlowSyncing ? "legacy initial sync" : "did start legacy incremental sync")"
+
+        WireLogger.sync.info(
+            message,
+            attributes: .legacySyncDidStartAttributes(
+                initialSync: isSlowSyncing
+            )
+        )
+    }
+
+    /// Logs the initial or incremental sync completion with the related duration
+    private func logSyncCompleted() {
+        let syncTotalDuration = syncTimeTracker.totalSyncDuration()
+        let formattedDuration = String(format: "%.2f", syncTotalDuration)
+        
+        let message = "did complete \(isSlowSyncing ? "legacy initial sync" : "legacy incremental sync") in \(formattedDuration)"
```

Comment:
suggestion: legacy info is in attributes, so no need to repeat it here
```suggestion
        let message = "\(isSlowSyncing ? "initial sync" : "incremental sync") completed"
```

Reply:
https://github.com/wireapp/wire-ios/pull/2965/commits/c34cf5e6c6115b0517e4422837745e09f5ded47d